addOns/websocket/src/main/java/org/zaproxy/zap/extension/websocket/pscan/WebSocketPassiveScannerManager.java

```diff
@@ -0,0 +1,252 @@
+    }
+
+    /**
+     * Add a passive scanner to thread safe list
+     *
+     * @return {@code true} if passive scanner is added to list successfully.
+     * @param passiveScanner the WebSocket Passive scan Plugin
+     */
+    private boolean addPlugin(WebSocketPassiveScannerDecorator passiveScanner) {
+        if (getPassiveScannersSet().contains(passiveScanner)) {
+            LOGGER.warn(
+                    "Insertion of "
+                            + passiveScanner.getName()
+                            + " is prevent in order to avoid the duplication");
+            return false;
+        }
+        return getPassiveScannersSet().add(passiveScanner);
+    }
+
+    /**
+     * Enables or disables all WebSocket Passive Scanners
+     *
+     * @param enabled {@code true} if the scanners should be enabled, {@code false} otherwise
+     */
+    public void setAllEnable(boolean enabled) {
+        Iterator<WebSocketPassiveScanner> iterator = getIterator();
+        while (iterator.hasNext()) {
+            ((WebSocketPassiveScannerDecorator) iterator.next()).setEnabled(enabled);
+        }
+    }
+
+    /**
+     * Enables or disables a WebSocket Passive Scanners
+     *
+     * @param enabled {@code true} if the scanner should be enabled, {@code false} otherwise
+     */
+    public void setEnable(WebSocketPassiveScanner scanner, boolean enabled) {
+
+        Iterator<WebSocketPassiveScanner> iterator = getIterator();
+        WebSocketPassiveScannerDecorator itScanner;
+        while (iterator.hasNext()) {
+            itScanner = ((WebSocketPassiveScannerDecorator) iterator.next());
+            if (itScanner.equals(scanner)) {
+                itScanner.setEnabled(enabled);
+                return;
+            }
+        }
+    }
+
+    /** Shut down the background thread if any. */
+    private void shutdownThread() {
+        if (this.passiveScanThread != null) {
+            passiveScanThread.shutdown();
+        }
+    }
+
+    /**
+     * Activating or Deactivating the passive scanning. That's not related with any {@link
+     * WebSocketPassiveScanner} just with background thread {@link WebSocketPassiveScanThread}
+     *
+     * @param activation if true activates the background thread
+     */
+    public void setThreadActivation(boolean activation) {
+        if (isThreadActive != activation) {
+            isThreadActive = activation;
+            if (activation) {
+                startThread();
+            } else {
+                shutdownThread();
+            }
+        }
+    }
+
+    public AlertManager getAlertManager() {
+        return alertManager;
+    }
+
+    /**
+     * Drop the passive scanner from the list
+     *
+     * @return {@code true} if passive scanner is dropped from list successfully.
+     */
+    public synchronized boolean removeScanner(WebSocketPassiveScanner passiveScanner) {
+        return getPassiveScannersSet().remove(passiveScanner);
```

Comment:
To work properly always the `passiveScanner` should be wrapped in a `WebSocketPassiveScannerDecorator`.

Reply:
Fixed in c4b9886